src/action_diff.rs

```diff
@@ -103,9 +104,9 @@ impl<A: Actionlike> SummarizedActionState<A> {
     }
 
     /// Captures the raw values for each action in the current frame.
-    pub fn summarize(
+    pub fn summarize<F: QueryFilter>(
```

Comment:
Are we allowed to put a default type here?

We should add a non-filtering `F` as the default if so, and document what one is if not.

Reply:
I'm new to Rust, would a default type param look like `pub fn summarize<F: QueryFilter = ()>`? That gives me:

```
defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions
this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
for more information, see issue #36887 <https://github.com/rust-lang/rust/issues/36887>
`#[deny(invalid_type_param_default)]` on by default
```

If it's not allowed, then let's rename the existing filtered implementation `summarize_filtered<F: QueryFilter>(...)`, and add a new `summarize(...)` that calls `summarize_filtered<()>()`.

Reply:
Yep, looks like it's not allowed. I agree with your suggestion :)